### ISSUE TITLE
Carbon : Select component selected value is string though we have integer in option values.

### ISSUE THREAD

**Issue by kavyanekkalapu** [data-driven-forms/react-forms#980]
**Scope**: <DESCRIBE SCOPE OF THIS ISSUE, i.e.: PF4 mapper, Form renderer>

**Description**

In schema, we have number as value in options, once, we selected options, it is giving out selected value as string.

<img width="630" alt="Screen Shot 2021-03-09 at 3 43 45 PM" src="https://user-images.githubusercontent.com/37085529/110535989-ee172c80-80ee-11eb-897d-c6399b8fd624.png">

**Expectation**

If we give integer as value in options, selected item should be type integer.

**Schema**

 ``` "fields": [
    {
      "component": "select",
      "label": "Select",
      "name": "select",
      "initialValue":2,
      "options": [
        {
          "label": "Dogs",
          "value": 1
        },
        {
          "label": "Cats",
          "value": 2
        },
        {
          "label": "Hamsters",
          "value": 3
        }
      ]
    }
  ];

**Comment by rvsia** [comment]
Hello, you can convert it to number by using a `datatype` attribute: https://data-driven-forms.org/schema/data-types

**Comment by kavyanekkalapu** [comment]
Hi @rvsia , unless i specifically mention dataType: 'number', it will convert value to string? is it a recent change? We did not have this issue with patternfly mapper, after switching to carbon mapper, issue came up.

**Comment by kavyanekkalapu** [comment]
PF4 mapper- here i did not mention dataType

<img width="654" alt="Screen Shot 2021-03-09 at 4 09 10 PM" src="https://user-images.githubusercontent.com/37085529/110538255-d7260980-80f1-11eb-86b8-df2efdcba666.png">

**Comment by Hyperkid123** [comment]
> Hi @rvsia , unless i specifically mention dataType: 'integer', it will convert value to string? is it a recent change? We did not have this issue with patternfly mapper, after switching to carbon mapper, issue came up.

@kavyanekkalapu the fact that it was working in PF3 mapper is caused by the PF3 component behavior. Same foes foe the PF4. Those libraries are using custom select components. The carbon select component is using a good old HTML select to handle the events. HTML select (and any other HTML form tag) value retrieved from `event.target.value` will be a string data type. There is no way around it.

Using the `dataType` prop will typecast the value to a number. Or you can try and open the issue in the carbon repo to preserve the data types.

**Comment by kavyanekkalapu** [comment]
@Hyperkid123 @rvsia This is the example we are trying https://stackblitz.com/edit/mtevuf?file=schema.js, it is not working with dataType: integer, if i don't select anything ( with initial value), it is returning integer, but if select something, than it is back to string, Could you please help us to fix the issue?

**Comment by rvsia** [comment]
@kavyanekkalapu thanks for reporting this! Actually the issue is in the number 0, if you change 0 to some other value, it works well. It will be probably some edge-case bug, I will investigate tomorrow.

**Comment by rvsia** [comment]
@kavyanekkalapu I found what the issue and made a PR to fix. We will also backport this fix to v2.

**Comment by kavyanekkalapu** [comment]
Thanks @rvsia for quick fix and PRs, looks like v2 backport pr build is failing, can you take a look please? we are using v2 and currently this issue is a blocker for us, Thanks again.

**Comment by rvsia** [comment]
@kavyanekkalapu The build is OK, it's just a codecoverage, not a big deal. However, my colleague is out for today and there is no one else who can merge/release it, so please be patient. I think it will be released by tomorrow.

There is a possible fix with using `parse` function: https://final-form.org/docs/react-final-form/types/FieldProps#parse

```jsx
      parse: value => {
        if (value === "0") return 0;
        return value;
      },
```

(however, that will put a warning to console that the html select is receiving an unsupported prop.. however, it's better than it doesn't work. 😏  )

**Comment by Hyperkid123** [comment]
@kavyanekkalapu the fix was released in `v2.24.1`. The version is intentionally marked as deprecated.
https://www.npmjs.com/package/@data-driven-forms/react-form-renderer/v/2.24.1

**Comment by kavyanekkalapu** [comment]
Thanks @Hyperkid123 @rvsia I verified the fix, it is working, closing the issue.
